framework/src/main/java/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -2444,77 +2468,46 @@ public AnnotatedTypeMirror getDefaultValueAnnotatedType(TypeMirror typeMirror) {
       WholeProgramInferenceJavaParserStorage.CallableDeclarationAnnos methodAnnos,
       List<AnnotationMirror> preconds) {
     List<AnnotationMirror> result = new ArrayList<>();
-    for (Map.Entry<VariableElement, AnnotatedTypeMirror> entry :
-        methodAnnos.getFieldToPostconditions().entrySet()) {
-      result.addAll(getPostconditionAnnotation(entry.getKey(), entry.getValue(), preconds));
+    for (Map.Entry<String, Pair<AnnotatedTypeMirror, AnnotatedTypeMirror>> entry :
+        methodAnnos.getPostconditions().entrySet()) {
+      result.addAll(
+          getPreOrPostconditionAnnotation(
+              entry.getKey(),
+              entry.getValue().first,
+              entry.getValue().second,
+              BeforeOrAfter.AFTER,
+              preconds));
     }
     Collections.sort(result, Ordering.usingToString());
     return result;
   }
 
   /**
-   * Return a list of precondition annotations for the given field. Returns an empty list if none
-   * can be created, because the qualifier has elements/arguments, which {@code @RequiresQualifier}
-   * does not support.
-   *
-   * <p>This is of the form {@code @RequiresQualifier(expression="this.elt",
-   * qualifier=MyQual.class)} when elt is declared as {@code @A} or {@code @Poly*} and f contains
-   * {@code @B} which is a sub-qualifier of {@code @A}.
-   *
-   * @param elt element for a field, which is declared in the same class as the method
-   * @param inferredType the type of the field, on method entry
-   * @return precondition annotations for the element (possibly an empty list)
-   */
-  public List<AnnotationMirror> getPreconditionAnnotation(
```

Comment:
This pull request removes the convenience methods `getPreconditionAnnotation` and `getPostconditionAnnotation`.  What is the reason for that?  I think it will make this pull request much easier to understand and review if that change is made in a separate pull request.

Reply:
There wasn't any reason to remove them, and I'm not really sure why I did, looking back. I've restored them.